torch/csrc/autograd/python_engine.cpp

```diff
@@ -131,7 +131,17 @@ PyObject *THPEngine_run_backward(THPEngine *self, PyObject *args, PyObject *kwar
 
     PyObject *grad = PyTuple_GET_ITEM(grad_tensors, i);
     if (THPVariable_Check(grad)) {
-      grads.push_back(((THPVariable*)grad)->cdata);
+      const Variable& grad_var = ((THPVariable*)grad)->cdata;
+#ifdef BUILD_NAMEDTENSOR
+      if (grad_var.has_names()) {
+        TORCH_WARN(
+            "Autograd was passed a named grad tensor with dims ", grad_var.names(),
+            ". Autograd does not yet support named tensor semantics, so all names ",
+            "will be ignored. In practice all computed gradients will still be correct "
+            "according to regular tensor semantics.");
+      }
```

Comment:
In the terminal state, I guess what you should have is the explicit grad does name matching with the output, and then you're off to the races. But this is fine for now :)

Reply:
Agreed. If I have time before the initial release I'll take a stab at doing this.